test/snippet/core/detail/all_of_in_type_list.cpp

```diff
@@ -24,26 +27,21 @@ int main()
         static_assert(std::is_same_v<id_t, std::type_identity<type>>, "id is of type std::type_identity<type>");
 
         if constexpr(std::is_same_v<type, bool>)
-            seqan3::debug_stream << "bool";
+            return true;
         else if constexpr(std::is_same_v<type, int>)
-            seqan3::debug_stream << "int";
+            return true;
         else if constexpr(std::is_same_v<type, float>)
-            seqan3::debug_stream << "float";
+            return true;
         else if constexpr(std::is_same_v<type, incomplete::type>)
-            seqan3::debug_stream << "incomplete::type";
-
-        seqan3::debug_stream << ", ";
+            return false;
     };
 
     // prints each type name, i.e. "int, float, bool, incomplete::type, \n"
-    seqan3::detail::for_each_type<int, float, bool, incomplete::type>(fn);
+    seqan3::debug_stream << std::boolalpha;
+    seqan3::debug_stream << seqan3::detail::all_of<seqan3::type_list<int, float, bool, incomplete::type>>(fn); // prints `false`.
```

Comment:
I think in this case `static_assert` would be much clearer. It made more since printing something for `for_each`